src/pages/v4/[chainName]/p/[projectId]/index.tsx

```diff
@@ -0,0 +1,82 @@
+import { FEATURE_FLAGS } from 'constants/featureFlags'
+import { OPEN_IPFS_GATEWAY_HOSTNAME } from 'constants/ipfs'
+import ReactQueryProvider from 'contexts/ReactQueryProvider'
+import { JBChainId, JBProjectProvider } from 'juice-sdk-react'
+import { useRouter } from 'next/router'
+import { V4ProjectDashboard } from 'packages/v4/views/V4ProjectDashboard/V4ProjectDashboard'
+import { wagmiConfig } from 'pages/v4/wagmiConfig'
+import React, { PropsWithChildren } from 'react'
+import { featureFlagEnabled } from 'utils/featureFlags'
+import {
+  arbitrumSepolia,
+  baseSepolia,
+  optimismSepolia,
+  sepolia,
+} from 'viem/chains'
+import { WagmiProvider } from 'wagmi'
+
+const chainNameMap: Record<string, JBChainId> = {
+  sepolia: sepolia.id,
+  opsepolia: optimismSepolia.id,
+  basesepolia: baseSepolia.id,
+  arbsepolia: arbitrumSepolia.id,
+}
+
+export default function V4ProjectPage() {
+  const router = useRouter()
+  const { chainName, projectId } = router.query
+  if (!chainName || !projectId) {
+    return <div>Invalid URL</div>
+  }
+  const chainId = chainNameMap[chainName as string]
+
+  return (
+    <_Wrapper>
+      <Providers chainId={chainId} projectId={BigInt(projectId as string)}>
+        <V4ProjectDashboard />
+      </Providers>
+    </_Wrapper>
+  )
+}
+
+const Providers: React.FC<
+  PropsWithChildren & { chainId: JBChainId; projectId: bigint }
+> = ({ chainId, projectId, children }) => {
+  return (
+    <ReactQueryProvider>
+      <WagmiProvider config={wagmiConfig}>
+        <JBProjectProvider
```

Comment:
JBProjectProvider loads up most of the project's data, and makes various contexts available.

Full list here: https://github.com/Bananapus/juice-sdk-v4/blob/351a46e13a42f3b0341575332a651c1d98aa817c/packages/react/src/contexts/JBProjectProvider/JBProjectProvider.tsx#L46-L56

Example: you should be able to get project metadata anywhere using:

```ts
const { metadata } = useJBProjectMetadataContext()
const { data, loading } = { metadata }
const projectName = data?.name
```

Throughout the SDK, most of the data is supplied with its loading state alongside (as shown above).